scripts/save.js

```diff
@@ -0,0 +1,129 @@
+// Description:
+//  This will help us save and do the saving through marvin
+//
+// Dependencies:
+//  request, lodash
+//
+// Configuration:
+//  None
+// Commands:
+//  marvin save - Try it out and see what marvin says
+
+var request = require('request');
+var _ = require('lodash')
+var slackToken = "cLSehYweAMPA1ZhgWMww9Xap";
+var allowedTimeSeperatorChars = [".", ":", "-"];
+
+var deniedMinutesResponses = [
+	'You cant trick me, pleae correct your time',
+	'You think im stupid? Adjust your input',
+	'Look at your time, its just wrong'
+];
+
+var errorResponses = [
+	'Ooopsi doopsie, I think i shit my leg',
+	'I dont think that packet came all the way to the end, sawwy',
+	'I failed master, let me make it up to you',
+	'Something went wrong and its all my fault',
+	'Pooop, i failed delivering that message'
+]
+
+var successResponses = [
+	'Yeaaahhjj, that task was saved!',
+	'Go figure, it worked! Ask Ken to se the result',
+	'Im not worthless afterall, it got saved, hurray!',
+	'Holy stinkballs, i made it! Its saved in the big internet',
+	'Whos the boss, i got it save for you master!'
+]
+
+var waitMessages = [
+	'just hang thight',
+	'take a coke and wait',
+	'make out with some girls while im finishing',
+	'grab a chair and look at the sky until im done'
+]
+
+module.exports = function (robot) {
+
+  robot.respond(/(save$)/i, function (msg) {
+    msg.send(
+    	"What would you like to save? \n"+
+    	"The last x lines: marvin save 5 lines \n"+
+    	"The last x minutes: marvin save 10 minutes \n"+
+    	"From x to y: marvin save 10:20 10:30"
+    );
+  });
+
+  robot.respond(/save (\d+) lines?$/i, function (msg) {
+  	saveRequest(msg, "LINE");
+  });
+
+  robot.respond(/save (\d+) minutes?$/i, function (msg) {
+  	saveRequest(msg, "MINUTES");
+  });
+
+  robot.respond(/save (([0-1]?[0-9]|[2][0-3]):([0-5][0-9])) (([0-1]?[0-9]|[2][0-3]):([0-5][0-9]))/i, function (msg) {
+  	if(parseInt(msg.match[1].split(":")[0]) > parseInt(msg.match[4].split(":")[0])) {
```

Comment:
Samme som over, er veldig vanskelig å se utifra koden hva som skjer (antar den splitter på `:` for å definere tiden, men dette burde kanskje gjøres mer eksplisitt (i en funksjon) eventuelt dokumenteres i en kommentar.